internal/provider/services/version_manager/version_manager.go

```diff
@@ -57,16 +57,16 @@ type VersionManagerService struct {
 }
 
 type Provider struct {
-	Name          string                    `json:"name" bson:"name" dynamodbav:"name"`
-	Version       string                    `json:"version" bson:"version" dynamodbav:"version"`
-	Protocols     []string                  `json:"protocols" bson:"protocols" dynamodbav:"protocols"`
-	Platforms     []*terrarium.PlatformItem `json:"platforms" bson:"platforms" dynamodbav:"platforms"`
-	Description   string                    `json:"description" bson:"description" dynamodbav:"description"`
-	SourceRepoUrl string                    `json:"source_repo_url" bson:"source_repo_url" dynamodbav:"source_repo_url"`
-	Maturity      string                    `json:"maturity" bson:"maturity" dynamodbav:"maturity"`
-	CreatedOn     string                    `json:"created_on" bson:"created_on" dynamodbav:"created_on"`
-	ModifiedOn    string                    `json:"modified_on" bson:"modified_on" dynamodbav:"modified_on"`
-	PublishedOn   string                    `json:"published_on" bson:"published_on" dynamodbav:"published_on"`
+	Name          string                    `json:"name,omitempty" bson:"name,omitempty" dynamodbav:"name,omitempty"`
+	Version       string                    `json:"version,omitempty" bson:"version,omitempty" dynamodbav:"version,omitempty"`
```

Comment:
Not all these should be set to omitempty - as in some cases if there is no name/version it would be an invalid entry and should fail.
Can you take a second look at these fields and confirm which ones are optional and which ones are required.

Reply:
sure, I reliazed that we might only need "omit empty" at modified on & published on field as they get added on different  conditions.